tests/functional/test_deploy.py

```diff
@@ -0,0 +1,127 @@
+import os
+import pytest
+import requests
+import subprocess
+import time
+
+# from juju.model import Model
+
+# Treat tests as coroutines
+pytestmark = pytest.mark.asyncio
+
+juju_repository = os.getenv('JUJU_REPOSITORY', '.').rstrip('/')
+series = ['xenial',
+          pytest.param('bionic', marks=pytest.mark.xfail(reason='canary')),
```

Comment:
Per comment above we should probably remove bionic from metadata until we've run canary functional tests for a while